source/php/Component/MegaMenu/megaMenu.blade.php

```diff
@@ -50,28 +50,46 @@ class="{{$baseClass}}__item {{$baseClass}}__item--parent o-grid-12 o-grid-6@md o
                         {{ $item['description'] }}
                     @endtypography
                 @endif
+
                 {{-- Children --}}
                 @if (!empty($item['children']))
-                    <ul class="{{$baseClass}}__sublist unlist u-margin__top--2">
+                    <ul class="{{$baseClass}}__sublist {{$baseClass}}__sublist--{{ ($childStyle ? 'flex' : 'list') }} unlist u-margin__top--2">
                         @foreach ($item['children'] as $child)
                             <li id="{{$id}}-item-{{$child['id']}}" class="{{$baseClass}}__item {{$baseClass}}__item--child {{ $child['classNames'] }}">
-                                @link([
-                                    'href' => $child['href'],
-                                    'classList' => [
-                                        $baseClass . '__link', 
-                                        $baseClass . '__link--child'
-                                    ]
-                                ])
-                                @if(isset($child['icon']) && !array_diff(['icon', 'size', 'classList'], array_keys($child['icon'])))
-                                    @icon([
-                                        'icon' => $child['icon']['icon'],
-                                        'size' => $child['icon']['size'],
-                                        'classList' => $child['icon']['classList'],
+                                @if(!empty($childStyle))
```

Comment:
Should probably check that it is a string being passed as well since it needs to be a string when being passed to button.